### ISSUE TITLE
BUG: _cython_blas.pyx:111:27: Converting to Python object not allowed without gil

### ISSUE THREAD

**Issue by sighingnow** [xprobe-inc/xorbits#251]
### Describe the bug

I'm encountering errors as follows with Python 3.11 and cython==0.29.26. I would like to confirm whether it is a known issue.

(Homebrew on MacOS has upgraded the default Python version to 3.11)

```python
xorbits/_mars/learn/utils/_cython_blas.pyx:111:20: Converting to Python object not allowed without gil

Error compiling Cython file:
------------------------------------------------------------
...

cdef void _rot(int n, floating *x, int incx, floating *y, int incy,
               floating c, floating s) nogil:
    """Apply plane rotation"""
    if floating is float:
        srot(&n, x, &incx, y, &incy, &c, &s)
                          ^
------------------------------------------------------------

xorbits/_mars/learn/utils/_cython_blas.pyx:111:27: Converting to Python object not allowed without gil

Error compiling Cython file:
------------------------------------------------------------
...

cdef void _rot(int n, floating *x, int incx, floating *y, int incy,
               floating c, floating s) nogil:
    """Apply plane rotation"""
    if floating is float:
        srot(&n, x, &incx, y, &incy, &c, &s)
                             ^
------------------------------------------------------------

xorbits/_mars/learn/utils/_cython_blas.pyx:111:30: Converting to Python object not allowed without gil

Error compiling Cython file:
------------------------------------------------------------
...

cdef void _rot(int n, floating *x, int incx, floating *y, int incy,
               floating c, floating s) nogil:
    """Apply plane rotation"""
    if floating is float:
        srot(&n, x, &incx, y, &incy, &c, &s)
```

**Comment by aresnow1** [comment]
We are planning to support Python 3.11 in the next release, but the work on this has not yet started. This should be a bug.

**Comment by aresnow1** [comment]
Are you interested in fixing this bug?

**Comment by sighingnow** [comment]
I would take a try.

**Comment by sighingnow** [comment]
The original error is caused by missing of scipy package.

The Python 3.11 compatibility issue with cython has been addressed in #252.
